nerdtree_plugin/vcs.vim

```diff
@@ -19,6 +20,26 @@ function! s:CreateTabTreeVCS(name)
     call g:NERDTreeCreator.createTabTree(empty(l:path) ? "" : l:path._str())
 endfunction
 
+" FUNCTION: s:ToggleTabTreeVCS(a:name) {{{1
+" Behaves the same as ToggleTabTree except roots directory at VCS root
+function! s:ToggleTabTreeVCS(name)
+    let l:path = g:NERDTreeCreator._pathForString(a:name)
+    let l:path = s:FindParentVCSRoot(l:path)
+    if g:NERDTree.ExistsForTab()
+        if !g:NERDTree.IsOpen()
+            call g:NERDTreeCreator._createTreeWin()
+            if !&hidden
+                call b:NERDTree.render()
+            endif
+            call b:NERDTree.ui.restoreScreenState()
+        else
+            call g:NERDTree.Close()
+        endif
+    else
+        call g:NERDTreeCreator.createTabTree(empty(l:path) ? "" : l:path._str())
+    endif
```

Comment:
Good addition. It looks like you've copied some code from `creator.vim`, and you can simplify this considerably by calling that function instead. Replace lines 28-40 with:

```vim
call g:NERDTreeCreator.toggleTabTree(empty(l:path) ? "" : l:path._str())
```